### ISSUE TITLE
Custom Edit Component example with nested data

### ISSUE THREAD

**Issue by arheinjohnson** [mui-org/material-ui-x#1726]
The custom edit component example: https://material-ui.com/components/data-grid/editing/#custom-edit-component

In this example the data isn't nested, such as { "Field": {"Subfield": int} }.
For implementation, I'm able to use this data structure for data grid by using valueGetter of GridColDef.
I'm trying to understand how to use api.commitCellChange along with valueGetter to use a nested data structure and a custom edit component. How do I do this?

## Order id 💳

#23360

**Comment by oliviertassinari** [comment]
In #1032 I was proposing the introduction of a `valueSetter` property to solve this problem.

Are you using a custom `renderEditCell` component?

Did you try following https://material-ui.com/components/data-grid/editing/#column-with-valuegetter?

**Comment by arheinjohnson** [comment]
Yes I'm already using valueGetter, and was not using renderEditCell, is that my problem?

**Comment by dtassone** [comment]
@arheinjohnson have you had a look at https://material-ui.com/components/data-grid/editing/#column-with-valuegetter ?

> Is it necessary to use commitCellChange, or can I instead update the state and have datagrid re-render?

No you could update the state directly, but it would be more complex  and commitCellChange does that for you.

**Comment by oliviertassinari** [comment]
@arheinjohnson any updates?

**Comment by arheinjohnson** [comment]
@oliviertassinari for now I'm using renderCell and updating state myself, so do not currently require this enhancement

**Comment by carl0s-sa** [comment]
Hi, I'm having the same issue here.
Could it be worth supporting propPaths in the `field` prop?

Example:
```js
const rows = [
  {
    productId: 0,
    prices: { EUR: 22.5, USD: 29.99 },
  },
]

const columns = [
  { field: 'prices.EUR', headerName: 'EUR price', editable: true },
  { field: 'prices.USD', headerName: 'USD price', editable: true },
  ....
]
```

I could pick this up if needed. I was snooping around in the code and I suppose one could change this here https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/hooks/features/rows/useGridParamsApi.ts#L137
to something like: `return R.path(field.split('.'), rowModel)` 
([Ramda.path dep for brevity](https://ramdajs.com/docs/#path))

Or maybe it's not that common of a case 😅 
What do you think?
